### ISSUE TITLE
Enum member, instead of the value of enum member, is given the type of the actual class

### ISSUE THREAD

**Issue by chazeon** [PyCQA/pylint#2263]
### Steps to reproduce

The result is from VSCode's pylint, writing a singleton with the assistance of `Enum`, is the recommended way of writing singleton pattern by [PEP 484](https://www.python.org/dev/peps/pep-0484/):

```python
class TestClass:
    def hello(self):
        return "Hello!"

class TestClassSingleton(enum.Enum):
    test_object = TestClass()

def get_test_object():
    return TestClassSingleton.test_object.value

test_class_instance = get_test_object()

print(test_class_instance.hello())
```

This code correctly prints `Hello!`.

### Current behavior

* The `test_class_instance` is of type “method 'value'”, and any call to it would result in `E1101`.
* `TestClassSingleton.test_object` is of type `TestClass`

![image](https://user-images.githubusercontent.com/2217102/42418454-49ec84fa-82d3-11e8-89f7-e7ae3e0bb3f9.png)

![image](https://user-images.githubusercontent.com/2217102/42418476-f4bec6b8-82d3-11e8-8bc5-5794a3b6aa4f.png)

### Expected behavior

* The `TestClassSingleton.test_object` should be of class `TestClassSingleton`.
* The `test_class_instance` should be of type `TestClass`.

### pylint --version output

```
tpylint 1.9.2,
astroid 1.6.5
Python 3.6.1 |Anaconda custom (64-bit)| (default, May 11 2017, 13:25:24) [MSC v.1900 64 bit (AMD64)]
```

**Comment by PCManticore** [comment]
I don't quite get what this issue is about. Where do you get these errors from, from `pylint`? If I run `pylint` at command line using the example you provided I get no errors. Can you paste a screenshot with the exact output you're getting from VSCode?

**Comment by PCManticore** [comment]
Thanks, now it makes more sense!

**Comment by Pierre-Sassoulas** [comment]
Closing because it cannot be reproduced in the latest pylint.
